tests/testthat/test-sector_profile.R

```diff
@@ -127,3 +127,51 @@ test_that("at company level, two matched and one unmatched products yield `value
   other <- pull(filter(out, !is.na(risk_category)), value)
   expect_equal(sort(other), c(0, 0, 2 / 3))
 })
+
+test_that("at product level, when `companies$tilt_sector` doesn't match `scenarios$sector` then `product$grouped_by` is `NA`", {
+  # This is what I think an analyst would say:
+  # > A product's tilt_sector with unmatched scenarios's sector yields a "missing benchmark"
+  #
+  # Translation to tiltIndicator's code:
+  # * given: the value "b" of `companies$clustered`
+  # * when: the corresponding value of `companies$tilt_sector` doesn't match any value of `scenarios$sector`
+  # * then: the result at product level has the value `NA` in `product$grouped_by`
+
+  # Based on this GoogleSheet row:
+  # https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064&range=A5:I5
+  # styler: off
+  companies <- tribble(
+    ~companies_id, ~clustered, ~activity_uuid_product_uuid, ~tilt_sector, ~tilt_subsector,       ~type,     ~sector,  ~subsector,
+              "a",        "b",                 "unmatched",  "unmatched",     "unmatched", "unmatched", "unmatched", "unmatched",
+  )
+  scenarios <- tribble(
+    ~sector,   ~subsector,  ~year, ~reductions, ~type, ~scenario,
+    "total",     "energy",   2050,         1.0, "ipr",       "a",
+  )
+  # styler: on
+
+  product <- sector_profile(companies, scenarios) |> unnest_product()
+
+  expect_true(is.na(product$grouped_by))
+})
+
+test_that("at product level, when `companies$tilt_sector` doesn't match `scenarios$sector` then `product$risk_category` and  `product$profile_ranking` are `NA`", {
+  # TODO: DRY with the test above
+  # styler: off
+  companies <- tribble(
+    ~companies_id, ~clustered, ~activity_uuid_product_uuid, ~tilt_sector, ~tilt_subsector,       ~type,     ~sector,  ~subsector,
+              "a",        "b",                 "unmatched",  "unmatched",     "unmatched", "unmatched", "unmatched", "unmatched",
+  )
+  scenarios <- tribble(
+    ~sector,   ~subsector,  ~year, ~reductions, ~type, ~scenario,
+    "total",     "energy",   2050,         1.0, "ipr",       "a",
+  )
+  # styler: on
+
+  product <- sector_profile(companies, scenarios) |> unnest_product()
+
+  # Tilman's expectations
+  # https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064&range=B19:D19
+  expect_true(is.na(product$risk_category))
+  expect_true(is.na(product$profile_ranking))
+})
```

Comment:
@Tilmon, 

This test is exactly like the one above but focuses on other expectations that you didn't mention in https://github.com/2DegreesInvesting/tiltIndicator/pull/739#issuecomment-1977426095 but are implicit from your [GoogleSheet](https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064). 

Later I may consolidate these expectations with the ones above, but for now they are a useful placeholder to discuss other results I didn't include in this test yet. 

Specifically, some values in the result differ from what you show in your GoogleSheet. Please see this reprex and let me know if I need to fix the code or if the current output is OK.

<details/>
<summary/>
reprex
</summary>

``` r
if (!interactive()) withr::local_options(width = 500)
devtools::load_all()
#> ℹ Loading tiltIndicator

companies <- tribble(
  ~companies_id, ~clustered, ~activity_uuid_product_uuid, ~tilt_sector, ~tilt_subsector,       ~type,     ~sector,  ~subsector,
            "a",        "b",                 "unmatched",  "unmatched",     "unmatched", "unmatched", "unmatched", "unmatched",
)
scenarios <- tribble(
  ~sector,   ~subsector,  ~year, ~reductions, ~type, ~scenario,
  "total",     "energy",   2050,         1.0, "ipr",       "a",
)
# styler: on

product <- sector_profile(companies, scenarios) |> unnest_product()
product
#> # A tibble: 1 × 11
#>   companies_id grouped_by risk_category profile_ranking clustered activity_uuid_product_uuid tilt_sector scenario  year type      tilt_subsector
#>   <chr>        <chr>      <chr>                   <dbl> <chr>     <chr>                      <chr>       <chr>    <dbl> <chr>     <chr>         
#> 1 a            <NA>       <NA>                       NA b         unmatched                  unmatched   <NA>        NA unmatched unmatched

# Different from the GoogleSheet
different <- c("scenario", "type", "tilt_subsector")
product |> select(all_of(different))
#> # A tibble: 1 × 3
#>   scenario type      tilt_subsector
#>   <chr>    <chr>     <chr>         
#> 1 <NA>     unmatched unmatched
```

</details>





For your convenience, here is a screenshot. The relevant row is the one highlighted.

![image](https://github.com/2DegreesInvesting/tiltIndicator/assets/5856545/1a6c4404-e612-4f7d-a41b-dd2993070c2d)